src/project_types/project.rs

```diff
@@ -154,20 +168,31 @@ impl Project {
         add: bool
     ) -> Result<(), Error> {
     
-        for mut u in &mut self.users {
-            if u.name == name {
-                if add {
-                    u.add_permission(perm);
-                } else {
-                    u.remove_permission(perm);
+        for i in 0..self.users.len() {
```

Comment:
Yeah, I know its correct, but it's like a for loop vs for-each loop. Between

```
for (int i = 0; i < array.length; i++) {
    process(array[i]);
}
```

and

```
for (Element e : array) {
     process(e);
}
```

one is a lot more readable.